GVFS/GVFS.Common/Git/GitAuthentication.cs

```diff
@@ -37,57 +38,62 @@ public bool IsBackingOff
 
         public bool IsAnonymous { get; private set; } = true;
 
-        public void ConfirmCredentialsWorked(string usedCredential)
+        public void ApproveCredentials(string credentialString)
         {
             lock (this.gitAuthLock)
             {
-                if (usedCredential == this.cachedAuthString)
+                // Don't reset the backoff if this is for a different credential than we have cached
+                if (credentialString == this.cachedCredentialString)
                 {
                     this.numberOfAttempts = 0;
                     this.lastAuthAttempt = DateTime.MinValue;
+
+                    // Tell Git to store the valid credential if we haven't already
+                    // done so for this cached credential.
+                    string username;
+                    string password;
+                    if (!this.isCachedCredentialStringApproved &&
+                        TryParseCachedAuthString(this.cachedCredentialString, out username, out password))

```

Comment:
@wilbaker `GitAuthentication` does not have an `ITracer` except ones passed in for some methods.

Should I update `GitAuthentication` so it requires an `ITracer` in the constructor or just add `ITracer` to the `Approve` and `Reject` methods?

Reply:
I think whichever option is easier.  If we already have an `ITracer` available when constructing `GitAuthentication` you could just add it to the constructor.

Reply:
Looking at passing an `ITracer` in via the constructor seemed like a lot of effort, since `GitAuthentication` is created inside the constructor of `Enlistment`, which would mean plumbing `ITracer` all the way through 😕 

I went with just taking `ITracer` in the `Approve` and `Reject` methods.